stdcxx_update_genode.patch

From: Christian Prochaska <christian.prochaska@genode-labs.com>

This patch is needed to generate the 'c++config.h' file when updating the
stdcxx library. See 'repos/libports/include/stdcxx/README for more details'.
---
 repos/ports/ports/gcc.hash                         |    2 -
 repos/ports/src/noux-pkg/gcc/patches/series        |    1 
 .../src/noux-pkg/gcc/patches/stdcxx_update.patch   |   80 ++++++++++++++++++++
 repos/ports/src/noux-pkg/gcc/target.inc            |    3 +
 4 files changed, 84 insertions(+), 2 deletions(-)
 create mode 100644 repos/ports/src/noux-pkg/gcc/patches/stdcxx_update.patch

diff --git a/repos/ports/ports/gcc.hash b/repos/ports/ports/gcc.hash
index 21ceee8..56fca2d 100644
--- a/repos/ports/ports/gcc.hash
+++ b/repos/ports/ports/gcc.hash
@@ -1 +1 @@
-092a36c078b00211af4dc29f1fbfcabfde8e8614
+2bc2b94894a9c25ef658c98382616770b5044a78
diff --git a/repos/ports/src/noux-pkg/gcc/patches/series b/repos/ports/src/noux-pkg/gcc/patches/series
index a90be6f..07b36e0 100644
--- a/repos/ports/src/noux-pkg/gcc/patches/series
+++ b/repos/ports/src/noux-pkg/gcc/patches/series
@@ -10,3 +10,4 @@ go_libbacktrace.patch
 target_libbacktrace.patch
 noux_build.patch
 arm.patch
+stdcxx_update.patch
diff --git a/repos/ports/src/noux-pkg/gcc/patches/stdcxx_update.patch b/repos/ports/src/noux-pkg/gcc/patches/stdcxx_update.patch
new file mode 100644
index 0000000..254e72a
--- /dev/null
+++ b/repos/ports/src/noux-pkg/gcc/patches/stdcxx_update.patch
@@ -0,0 +1,80 @@
+stdcxx_update.patch
+
+From: Christian Prochaska <christian.prochaska@genode-labs.com>
+
+
+---
+ Makefile.tpl              |    1 +
+ configure.ac              |    2 ++
+ libstdc++-v3/acinclude.m4 |   16 ++++++++--------
+ libstdc++-v3/configure.ac |    2 +-
+ 4 files changed, 12 insertions(+), 9 deletions(-)
+
+diff --git a/Makefile.tpl b/Makefile.tpl
+index a89f815..c326e87 100644
+--- a/Makefile.tpl
++++ b/Makefile.tpl
+@@ -506,6 +506,7 @@ COMPILER_AS_FOR_TARGET=@COMPILER_AS_FOR_TARGET@
+ COMPILER_LD_FOR_TARGET=@COMPILER_LD_FOR_TARGET@
+ COMPILER_NM_FOR_TARGET=@COMPILER_NM_FOR_TARGET@
+ 
++CPPFLAGS_FOR_TARGET = @CPPFLAGS_FOR_TARGET@
+ CFLAGS_FOR_TARGET = @CFLAGS_FOR_TARGET@
+ CXXFLAGS_FOR_TARGET = @CXXFLAGS_FOR_TARGET@
+ 
+diff --git a/configure.ac b/configure.ac
+index dbfaf4d..5a2d3a4 100644
+--- a/configure.ac
++++ b/configure.ac
+@@ -2380,6 +2380,8 @@ AC_ARG_WITH([debug-prefix-map],
+   [DEBUG_PREFIX_CFLAGS_FOR_TARGET=])
+ AC_SUBST(DEBUG_PREFIX_CFLAGS_FOR_TARGET)
+ 
++AC_SUBST(CPPFLAGS_FOR_TARGET)
++
+ # During gcc bootstrap, if we use some random cc for stage1 then CFLAGS
+ # might be empty or "-g".  We don't require a C++ compiler, so CXXFLAGS
+ # might also be empty (or "-g", if a non-GCC C++ compiler is in the path).
+diff --git a/libstdc++-v3/acinclude.m4 b/libstdc++-v3/acinclude.m4
+index ce44a6a..2373dd5 100644
+--- a/libstdc++-v3/acinclude.m4
++++ b/libstdc++-v3/acinclude.m4
+@@ -1136,17 +1136,17 @@ AC_DEFUN([GLIBCXX_ENABLE_C99], [
+     fi
+ 
+     # Option parsed, now set things appropriately.
+-    if test x"$glibcxx_cv_c99_math_cxx98" = x"no" ||
+-       test x"$glibcxx_cv_c99_complex_cxx98" = x"no" ||
+-       test x"$glibcxx_cv_c99_stdio_cxx98" = x"no" ||
+-       test x"$glibcxx_cv_c99_stdlib_cxx98" = x"no" ||
+-       test x"$glibcxx_cv_c99_wchar_cxx98" = x"no"; then
+-      enable_c99=no;
+-    else
++#    if test x"$glibcxx_cv_c99_math_cxx98" = x"no" ||
++#       test x"$glibcxx_cv_c99_complex_cxx98" = x"no" ||
++#       test x"$glibcxx_cv_c99_stdio_cxx98" = x"no" ||
++#       test x"$glibcxx_cv_c99_stdlib_cxx98" = x"no" ||
++#       test x"$glibcxx_cv_c99_wchar_cxx98" = x"no"; then
++#      enable_c99=no;
++#    else
+       AC_DEFINE(_GLIBCXX_USE_C99, 1,
+         [Define if C99 functions or macros from <wchar.h>, <math.h>,
+         <complex.h>, <stdio.h>, and <stdlib.h> can be used or exposed.])
+-    fi
++#    fi
+ 
+     gcc_no_link="$ac_save_gcc_no_link"
+     LIBS="$ac_save_LIBS"
+diff --git a/libstdc++-v3/configure.ac b/libstdc++-v3/configure.ac
+index 088f5e1..b8addaf 100644
+--- a/libstdc++-v3/configure.ac
++++ b/libstdc++-v3/configure.ac
+@@ -209,7 +209,7 @@ AC_CHECK_HEADERS(fenv.h complex.h)
+ GLIBCXX_CHECK_C99_TR1
+ 
+ # For the EOF, SEEK_CUR, and SEEK_END integer constants.
+-#GLIBCXX_COMPUTE_STDIO_INTEGER_CONSTANTS
++GLIBCXX_COMPUTE_STDIO_INTEGER_CONSTANTS
+ 
+ # For gettimeofday support.
+ GLIBCXX_CHECK_GETTIMEOFDAY
diff --git a/repos/ports/src/noux-pkg/gcc/target.inc b/repos/ports/src/noux-pkg/gcc/target.inc
index 8ba3251..b388d45 100644
--- a/repos/ports/src/noux-pkg/gcc/target.inc
+++ b/repos/ports/src/noux-pkg/gcc/target.inc
@@ -14,7 +14,6 @@ CONFIGURE_ARGS = --program-prefix=$(PROGRAM_PREFIX) \
                  --with-gnu-ld \
                  --disable-tls \
                  --disable-threads \
-                 --disable-hosted-libstdcxx \
                  --enable-shared \
                  --enable-multiarch \
                  --disable-sjlj-exceptions
@@ -28,6 +27,8 @@ ENV += CC_FOR_TARGET=$(CC) CXX_FOR_TARGET=$(CXX) GCC_FOR_TARGET=$(CC) CPP_FOR_TA
 # libgcc does not evaluate CPPFLAGS_FOR_TARGET, so everything is put into CFLAGS_FOR_TARGET here
 ENV += CFLAGS_FOR_TARGET='-I$(BASE_DIR)/../../tool -DUSE_PT_GNU_EH_FRAME -Dinhibit_libc -fPIC'
 
+ENV += CPPFLAGS_FOR_TARGET='$(CPPFLAGS)'
+
 # libsupc++
 ENV += CXXFLAGS_FOR_TARGET='-fPIC'
 
